azext_iot/common/utility.py

```diff
@@ -344,12 +346,18 @@ def unpack_pnp_http_error(e):
 
 def unpack_msrest_error(e):
     """Obtains full response text from an msrest error"""
+    from typing import Callable
 
     op_err = None
     try:
-        op_err = json.loads(e.response.text)
+        err_txt = ""
+        if isinstance(e.response.text, Callable):
+            err_txt = e.response.text()
```

Comment:
interesting - are the adu error texts callable?

Reply:
This is a result of the latest autorest python generator, embracing more track 2 patterns + integration with azure core.